src/vis/vis.js

```diff
@@ -215,46 +187,20 @@ var Vis = View.extend({
   },
 
   load: function (data, options) {
-
-    var self = this;
-    this.https = (window && window.location.protocol && window.location.protocol === 'https:') || !!data.https;
-
-    // Load the viz.json in case we receive a url instead of a JSON object
     if (typeof (data) === 'string') {
       var url = data;
-
       Loader.get(url, function (data) {
         if (data) {
-          self.load(data, options);
+          this.load(data, options);
         } else {
-          self.throwError('error fetching viz.json file');
+          this.throwError('error fetching viz.json file');
```

Comment:
Nevermind this comment, just saw the `.bind(this)` below… (will delete comment…)